nextcloud/mail
# fix: Fix sending local draft

Fixes https://github.com/nextcloud/mail/issues/8631
Fixes https://github.com/nextcloud/mail/issues/8630

Two commits:
1) Harden logic so the draft service only handles messages that are type=draft, outbox service messages that are type=outgoing. It doesn't fix a lot. It just makes the logic errors more obvious.
2) Fix the logic so that drafts are properly converted to outbox messages before they are sent.

## How to test

Three workflows:

1) The fast typer: compose new message but send right away. Be fast. There should not be a request to save the message as draft.
2) The relaxed typer: compose a new message but take your time. Let there be at least one draft, only the send.
3) The time traveler: composer a message and schedule it for later. Open the mailbox, make modifications and save/send.
